test-suite/misc/tc_declaration_observer/observer.ml

```diff
@@ -0,0 +1,19 @@
+
+let o = open_out "main.out"
+
+let observe x =
+  let open Classes in
+  let open Typeclasses in
+  let open Hints in
+  let p = Pp.string_of_ppcmds in
+  match x with
+  | NewClass { cl_impl } ->
+      Printf.fprintf o "NewClass %s\n" (p (Printer.pr_global cl_impl))
+  | NewInstance { instance ; info = { hint_priority }; locality } ->
+      Printf.fprintf o "NewInstance %s %s %s\n"
```

Comment:
```suggestion
      Printf.fprintf o "NewInstance %s%s %s\n"
```
pr_opt includes a leading space when non nil (you can see the resulting double space in the reference file)

Reply:
It is a test file, do we really care about a double space in the output?